### ISSUE TITLE
Typescript $group compilation error

### ISSUE THREAD

**Issue by igor-juro** [Automattic/mongoose#11067]
In aggregation pipeline in `$group` step when you use both `_id` (string) and any other key, typescript throws an error.
Example:
```
const model = mongoose.model("Model", new mongoose.Schema());

  const data = model.aggregate([
    {
      $match: {
        somevalue: "string"
      }
    },
    {
      $group: {
        _id: "$somevalue",
        types: { $addToSet: "$somevalue" }
      }
    }
  ]);
```
Error
```
Type '{ _id: string; types: { $addToSet: string; }; }' is not assignable to type '{ [key: string]: { $min?: any; $max?: any; $addToSet?: any; $push?: any; $count?: any; $accumulator?: any; $avg?: any; $first?: any; $last?: any; $mergeObjects?: any; $stdDevPop?: any; $stdDevSamp?: any; $sum?: any; }; _id: any; }'.
  Property '_id' is incompatible with index signature.
    Type 'string' has no properties in common with type '{ $min?: any; $max?: any; $addToSet?: any; $push?: any; $count?: any; $accumulator?: any; $avg?: any; $first?: any; $last?: any; $mergeObjects?: any; $stdDevPop?: any; $stdDevSamp?: any; $sum?: any; }'.ts(2322)
```
tsconfig
```
{
  "compilerOptions": {
    "strict": true,
    "module": "commonjs",
    "jsx": "preserve",
    "esModuleInterop": true,
    "sourceMap": true,
    "allowJs": true,
    "lib": [
      "es6",
      "dom"
    ],
    "rootDir": "src",
    "moduleResolution": "node"
  }
}
```
You can see a code sandbox [here](https://codesandbox.io/s/ecstatic-mirzakhani-vsm61?file=/src/index.ts) 
Apparently, in mongoose typings in `Group` interface
```
export interface Group {
      /** [`$group` reference](https://docs.mongodb.com/manual/reference/operator/aggregation/group) */
      $group: {
        _id: any;
        [key: string]: { [op in AccumulatorOperator]?: any }
      }
    }
```
[key: string] overrides `_id`

**Comment by rpenido** [comment]
Duplicata of #11059 ?

**Comment by DavideViolante** [comment]
This can be safely closed since fixed in mentioned PR ^ released in v6.1.3.
Not sure why it didn't close automatically.
